### PR TITLE
oci: layout: print mime type received when cannot convert manifest

### PR DESCRIPTION
**Pull request by runcom**
Fix https://github.com/projectatomic/skopeo/issues/198

I'll revendor in skopeo once this is merged. @mtrmac PTAL

Signed-off-by: Antonio Murdaca <runcom@redhat.com>

### CODE REVIEW

File: oci/layout/oci_dest.go [hunk 1]
```diff
@@ -133,7 +133,7 @@ func createManifest(m []byte) ([]byte, string, error) {
 	case imgspecv1.MediaTypeImageManifest:
 		return m, mt, nil
 	}
-	return nil, "", fmt.Errorf("Unrecognized manifest media type")
+	return nil, "", fmt.Errorf("Unrecognized manifest media type: %q", mt)
```
Comment (mtrmac) [inline oci/layout/oci_dest.go]: ACK, thanks.

File: oci/layout/oci_dest_test.go [hunk 2]
```diff
@@ -58,3 +58,19 @@ func TestPutBlobDigestFailure(t *testing.T) {
...
+
+	err = ociDest.PutManifest([]byte(m))
+	require.Error(t, err)
+	assert.Contains(t, `Unrecognized manifest media type: "application/vnd.docker.distribution.manifest.v1+prettyjws"`, err.Error())
```
Comment (mtrmac) [inline oci/layout/oci_dest_test.go]: I suppose this identifies the underlying bug in https://github.com/projectatomic/skopeo/issues/198 : we guess schema1 manifests as a MIME type which isn’t recognized.

OTOH as soon as we will recognize that MIME type, this test will permanently break…

**Comment by mtrmac** [comment]
LGTM though unless I am missing something, the test will be removed shortly. Feel free to merge either with it or without it.

File: oci/layout/oci_dest_test.go (hunk 2 above)
Comment (runcom) [inline oci/layout/oci_dest_test.go]: > this test will permanently break…

yep, that'll be the time we'll update it I guess but right now this is tracking a specific bug

**Comment by runcom** [comment]
LGTM

> LGTM though unless I am missing something, the test will be removed shortly. Feel free to merge either with it or without it.

We'll updated this test with another manifest and unrecognized mime type to make it functional again once we add schema1 support (if?)

File: oci/layout/oci_dest_test.go [hunk 3]
```diff
@@ -58,3 +58,19 @@ func TestPutBlobDigestFailure(t *testing.T) {
...
...\"Cmd\":null,\"Image\":\"\",\"Volumes\":null,\"WorkingDir\":\"\",\"Entrypoint\":null,\"NetworkDisabled\":false,\"MacAddress\":\"\",\"OnBuild\":null,\"Labels\":null},\"docker_version\":\"1.6.2\",\"config\":{\"Hostname\":\"\",\"Domainname\":\"\",\"User\":\"\",\"Memory\":0,\"MemorySwap\":0,\"CpuShares\":0,\"Cpuset\":\"\",\"AttachStdin\":false,\"AttachStdout\":false,\"AttachStderr\":false,\"PortSpecs\":null,\"ExposedPorts\":null,\"Tty\":false,\"OpenStdin\":false,\"StdinOnce\":false,\"Env\":null,\"Cmd\":null,\"Image\":\"\",\"Volumes\":null,\"WorkingDir\":\"\",\"Entrypoint\":null,\"NetworkDisabled\":false,\"MacAddress\":\"\",\"OnBuild\":null,\"Labels\":null},\"architecture\":\"amd64\",\"os\":\"linux\",\"Size\":9356886}\n"}],"schemaVersion":1,"signatures":[{"header":{"jwk":{"crv":"P-256","kid":"SVJ4:Q6G3:SXTN:H6LT:7PXH:DHUZ:SGTB:5TMV:YPIV:UPHY:MRHO:PN6V","kty":"EC","x":"qrSsA2UAKEFlDhLk12zoWpnHgYcTNfEOWGZU46pzhfk","y":"RtD_vGFtagPlheiunLvZL02LOssnu7DqShuBwc6Ml44"},"alg":"ES256"},"signature":"YzfU_rKQLWqG74uilltTiV3O92lfEjaG5wJkVt_dCtjH_C5AeghfQttnbtceJOyiaU7xP2yEnjdultutsxkQKQ","protected":"eyJmb3JtYXRMZW5ndGgiOjI4NDgsImZvcm1hdFRhaWwiOiJDbjAiLCJ0aW1lIjoiMjAxNi0wOS0xMFQwODoyMDowOFoifQ"}]}`
```
Comment (mtrmac) [inline oci/layout/oci_dest_test.go]: Please use fixture files for these huge constants in the future.

**Pull request merged**